core/src/main/java/org/jdbi/v3/core/result/ResultIterable.java

```diff
@@ -50,27 +52,22 @@
      * @param mapper   row mapper
      * @param ctx      statement context
      * @param <T>      the mapped type
+     * @deprecated the ResultIterable type is not intended to be constructed by user code
      * @return the result iterable
      */
+    @Deprecated
     static <T> ResultIterable<T> of(Supplier<ResultSet> resultSetSupplier, RowMapper<T> mapper, StatementContext ctx) {
-        return () -> {
-            try {
-                ResultSet resultSet = resultSetSupplier.get();
-                ctx.addCleanable(resultSet::close);
-
-                return new ResultSetResultIterator<>(resultSet, mapper, ctx);
-            } catch (SQLException e) {
-                throw new ResultSetException("Unable to iterate result set", e, ctx);
-            }
-        };
+        return new ResultSetResultIterable<>(mapper, ctx, resultSetSupplier, null);
```

Comment:
Because in this case we don't have the static element type at all. Originally, I treated it as a collection of just `List` (raw type) but that doesn't work because all the collector factories are coded and tested to not work with raw types. So I had to fudge it, or change the way the collector factories work.